tests/semantic_analyser.cpp

```diff
@@ -2804,6 +2826,11 @@ kprobe:f { $x = -1; $x = 10223372036854775807; }
 stdin:1:35-56: ERROR: Tuple type mismatch: (int64,uint32) != (int64,int32).
 kprobe:f { $x = (0, (uint32)123); $x = (0, (int32)-123); }
                                   ~~~~~~~~~~~~~~~~~~~~~
+)");
+  test_error("kprobe:f { $x = (uint32)5; $x += 1; }", R"(
```

Comment:
This should not be an error IMHO. We are able to tell statically that `1` is within the range of an unsigned 32 bit int and case it to the correct integer size (this is what we do for variable assignments).

Reply:
I agree this ideally should be supported. But the issue is the "can fit" checks are all behind `assignment.expr->is_literal`. And while this looks like a literal, it's actually not. The parser desugars this to `AssignVarStatement(Variable, Binop(Variable, Integer(1)))`. So we've lost that data. Unclear to me if it can be easily recovered. Maybe worth a try? But gut feeling is it may  not be worthwhile.

Reply:
> And while this looks like a literal, it's actually not.

Ah that's a bummer. Ok, I won't block on this then. I'll also poke around and see if I can make integers more flexible: https://github.com/bpftrace/bpftrace/issues/3518

Reply:
Gonna move this to a commented out "positive" test. Having a test like this encodes wrong intention. It'll seem to suggest we want it this way. Rather than it being a limitation.